src/paradigma/tremor/tremor_analysis.py

```diff
@@ -101,4 +106,99 @@ def detect_tremor_io(input_path: Union[str, Path], output_path: Union[str, Path]
     metadata_time.channels = [config.time_colname]
     metadata_time.units = ['relative_time_ms']
 
-    write_df_data(metadata_time, metadata_samples, output_path, 'tremor_meta.json', df)
\ No newline at end of file
+    write_df_data(metadata_time, metadata_samples, output_path, 'tremor_meta.json', df)
+
+
+def aggregate_tremor_power(tremor_power: pd.Series, config: TremorQuantificationConfig):
+
+    tremor_power = np.log10(tremor_power+1) # convert to log scale
+    
+    # calculate median and 90th percentile of tremor power
+    tremor_power_median = tremor_power.median()
+    tremor_power_90th_perc = tremor_power.quantile(config.percentile_tremor_power)
+    
+    # calculate modal tremor power
+    bin_edges = np.linspace(0, 6, 301)
+    kde = gaussian_kde(tremor_power)
+    kde_values = kde(bin_edges)
+    max_index = np.argmax(kde_values)
+    tremor_power_mode = bin_edges[max_index]
+
+    return tremor_power_median, tremor_power_90th_perc, tremor_power_mode
+
+
+def quantify_tremor(df: pd.DataFrame, config: TremorQuantificationConfig, start_time: datetime) -> pd.DataFrame:
+
+    time = [start_time + timedelta(seconds=x) for x in df.time]
+
+    # determine valid days
+    daytime_hours = range(config.daytime_hours_lower_bound,config.daytime_hours_upper_bound)
+    nr_windows_per_day = Counter([dt.date() for dt in time if dt.hour in daytime_hours])
+    valid_days = [day for day, count in nr_windows_per_day.items() 
+                  if count * config.window_length_s / 3600 >= config.valid_day_threshold_hr]
+    df_aggregates = pd.DataFrame({'nr_valid_days': [len(valid_days)]}) # save number of valid days
```

Comment:
Waarom is dit een list?

Reply:
Ik krijg een error als ik de vierkante haken weglaat, misschien omdat het een kolom moet worden in de dataframe?